analyzers/src/SonarAnalyzer.CSharp/Rules/MagicNumberShouldNotBeUsed.cs

```diff
@@ -96,5 +107,11 @@ private static bool IsInsideProperty(SyntaxNode node)
             var parent = node.Parent;
             return parent is ReturnStatementSyntax || parent is EqualsValueClauseSyntax;
         }
+
+        private static bool ToStringContainsAnyAcceptedNames(SyntaxNode syntaxNode)
+        {
+            var toString = syntaxNode.ToString().ToLower();
```

Comment:
My understanding is that we accept the methods `Count`, `Size`, `Length`, `Order` coming from particular namespaces (e.g., `IEnumerable`). 
In this case, the full name needs to be checked otherwise if a developer creates their own method `count` which might do something different, it's going to be a false negative.
Also, I think the character case needs to be taken into account (no need to turn the method names to lower), as `COUNT()` and `Count()` are two different methods even if they are in the same class.

Reply:
In case the intention is to avoid using the semantic model, at least a strict comparison with equals will create fewer FNs IMO.

Reply:
well, I'd say that in general if we want to compare anything that is Count to a single digit, it should be fine

`if (Foo.BarCount < 3) {}`

why would the rule ask the dev to put 3 in a magic constant?

Reply:
I think the risk would be something like `if (CountOfMonteChristo == 9)`, or something like this, but I'd say it's fairly low

And I dropped the idea of `Order` - should I actually add it?

Will people do something like `if (Foo.Order < 3)` - I guess I should?

Reply:
(offline discussion - the initial idea was to avoid noise when someone is comparing the size of a collection with single-digit number to access the first elements - so I reduced it to this scenario

we will still have FPs because this is a simple AST-based rule so we won't have dataflow and we don't plan to 
we will still have FPs , in general, with this rule - this is why it's not (and probably never will be) part of SonarWay)